dut/drivers.wake

```diff
@@ -0,0 +1,107 @@
+
+# create a BasicDriverPlan
+# duhFile => path to your duh file (device.json5)
+# OMFile => a function to produce the OMFile, takes Unit as an arg
+# VendorName => the vendor name used for the device
+# DeviceName => the name of the device
+# LocalMetalDirectory => the metal/drivers directory in the device's repository
+global def makeBasicDriverPlan duhFile omFile vendorName deviceName localMetalDirectory dutName =
+  BasicDriverPlan duhFile omFile vendorName deviceName localMetalDirectory dutName True False
+
+# This function provides an external interface through the publish/subscribe
+# variable 'basicDriverPlans' If you want automatic basic driver creation,
+# create a BasicDriverPlan using makeBasicDriverPlan, Then build the drivers
+# by calling createBasicDriver
+def getDevicePlan dutName =
+  def plans = subscribe basicDriverPlans
+  def samePlan aPlan =
+    aPlan.getBasicDriverPlanDUTName ==~ dutName
+  find samePlan plans
+  | omap getPairFirst
+
+def basicPlanToPlan basicDriverPlan =
+  def omfile = basicDriverPlan.getBasicDriverPlanOMFile Unit
+  def extraFlags =
+    def overwrite =
+      if basicDriverPlan.getBasicDriverPlanOverwriteExisting
+      then "--overwrite-existing", Nil
+      else Nil
+    def onlyBaseHeader =
+      if basicDriverPlan.getBasicDriverPlanOnlyBaseHeader
+      then "--base-header-only", Nil
+      else Nil
+    overwrite ++ onlyBaseHeader
+  def cmd =
+    "{here}/../scripts/generate_drivers.py".simplify,
+    "--object-model",
+    omfile.getPathName,
+    "--duh-document",
+    basicDriverPlan.getBasicDriverPlanDUHDocument.getPathName,
+    "--vendor",
+    basicDriverPlan.getBasicDriverPlanVendorName,
+    "--device",
+    basicDriverPlan.getBasicDriverPlanDeviceName,
+    "--metal-dir",
+    basicDriverPlan.getBasicDriverPlanLocalMetalDirectory,
+    extraFlags
+  def visibleFiles =
+    "{here}/../scripts/generate_drivers.py".simplify.source,
+    basicDriverPlan.getBasicDriverPlanDUHDocument,
+    omfile,
+    basicDriverPlan.getBasicDriverPlanLocalMetalDirectory.source,
+    Nil
+  makePlan cmd visibleFiles
+  | addPythonEnv ("{here}/../scripts/generate_drivers_env".simplify)
+
+tuple DriverOutput =
+  global CFiles:      List Path
+  global IncludeDirs: List String
+
+def foo x =
+  def bar = format x | println
+  x
+
+def makeDriverOutput basicDriverPlan jobOutputs =
+  def ifPathCFile path =
+    path.getPathName
+    | matches `.*\.[c,h]`
+  def cFiles = filter ifPathCFile jobOutputs
+  def metalDir = basicDriverPlan.getBasicDriverPlanLocalMetalDirectory
+  def deviceName = basicDriverPlan.getBasicDriverPlanDeviceName
+  def includeDirs =
+    metalDir,
+    Nil
+  DriverOutput cFiles includeDirs
+
+
+def makeBasicDrivers basicDriverPlan =
+  basicPlanToPlan basicDriverPlan
+  | runJob
+  | getJobOutputs
+  | makeDriverOutput basicDriverPlan
+
+# External wake target to create basic drivers
+# deviceName => the name of the DUT plan
+global def createBasicDriver dutPlanName =
```

Comment:
The comment says the argument name is `deviceName`, but the actual code says the argument is `dutPlanName`. The former makes more sense to me, but it appears that your code is actually making use of the latter. Why is it that we are filtering based on the DUT name rather than the device name?

Reply:
We need to build the DUT to get the OM to feed into the driver creation. Comment is wrong. 

I think this will change, so I will put off fixing it for a bit.